### DISCUSSION TITLE
Kubernetes runtime vs registry

### DISCUSSION THREAD

**Discussion by Davincible** [micro/go-micro#2565] [General] [answered]
@asim how do the k8s runtime and registry differ exactly?

I've been running with default runtime and k8s registry perfectly fine, but what extra does the k8s runtime offer?

**Comment by asim** [comment, marked as answer]
The runtime package is for quite literally running code. So the k8s runtime will run services. The registry is purely service discovery so there's still a need to actually manage deployment. The runtime package was used by the micro runtime. It's not something a service would directly import or use itself.

**Reply by Davincible** [reply]
So there's no benefit to my services using the k8s runtime?

**Reply by asim** [reply]
No I don't believe so unless you want to write a self deploying service which in itself is a very interesting concept.

**Reply by Davincible** [reply]
Definitely, although arguably not needed with k8s scheduler I guess

**Reply by asim** [reply]
My original goal was to implement self replicating services. So if a service could be deployed and had access to the k8s runtime it could autoscale based on business logic rather than generic metrics. Now I think it's more just useful for simpler access to k8s programmatically since k8s is quite bloated and complex.

**Reply by Davincible** [reply]
Ah I see. Yeah that makes sense. K8s access could definitely be useful in some cases.
